packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -53,76 +72,72 @@ export class SCWSigner implements Signer {
         params: this.metadata,
       },
     });
-    const response: RPCResponseMessage = await this.communicator.postRequestAndWaitForResponse(
-      handshakeMessage
-    );
+    const response: RPCResponseMessage =
+      await this.communicator.postRequestAndWaitForResponse(handshakeMessage);
 
     // store peer's public key
     if ('failure' in response.content) throw response.content.failure;
     const peerPublicKey = await importKeyFromHexString('public', response.sender);
     await this.keyManager.setPeerPublicKey(peerPublicKey);
 
     const decrypted = await this.decryptResponseMessage<AddressString[]>(response);
-    this.updateInternalState({ method: 'eth_requestAccounts' }, decrypted);
 
-    const result = decrypted.result;
-    if ('error' in result) throw result.error;
+    if ('error' in decrypted.result) throw decrypted.result.error;
 
-    return this.stateManager.accounts;
+    this._accounts = decrypted.result.value;
+    this.storage.storeObject(ACCOUNTS_KEY, this._accounts);
+    this.updateListener.onAccountsUpdate(this._accounts);
+    return this._accounts;
```

Comment:
instead of having separate call `this.updateInternalState({ method: 'eth_requestAccounts' }, decrypted);`
persistency and state update are handled directly here.